### PR TITLE
Make trilinos-solvers optional.

### PR DESCRIPTION
**Pull request by alanw0**
Removed explicit activation of belos, muelu, amesos2, ifpack2 Added infrastructure for enabling/disabling them.
Also make ~trilinos-solvers default on crusher.

### CODE REVIEW

File: configs/crusher/packages.yaml [hunk 1]
```diff
@@ -97,6 +97,8 @@ packages:
           - openblas/0.3.17
   trilinos:
     require: "@develop"
+  nalu-wind:
+    variants: ~trilinos-solvers
   all:
```
Comment (PaulMullowney) [inline configs/crusher/packages.yaml:102]: Does this mean it's off by default? And so matrix free will also get turned off in the nalu build?

Comment (psakievich) [reply configs/crusher/packages.yaml:102]: Yes but just for crusher

**Review by psakievich — APPROVED** [review]

File: configs/crusher/packages.yaml (hunk 1 above)
Comment (alanw0) [reply configs/crusher/packages.yaml:102]: Yes but here it only applies to crusher. What do you think? I was thinking we usually don't want trilinos-solvers built on crusher, but I can easily drop this if you think it should remain on by default.

Comment (PaulMullowney) [reply configs/crusher/packages.yaml:102]: Fine by me

**Review by PaulMullowney — APPROVED** [review]

File: configs/crusher/packages.yaml (hunk 1 above)
Comment (alanw0) [reply configs/crusher/packages.yaml:102]: Now that I think about this, if we are going to turn trilinos-solvers off by default, we should also turn hypre on by default. Is it on or off by default?

Comment (tasmith4) [reply configs/crusher/packages.yaml:102]: pretty sure it is now on by default after our latest spack change

Comment (alanw0) [reply configs/crusher/packages.yaml:102]: Yes it appears so... Did we do that deliberately? Maybe we need an opinion from @PaulMullowney on whether hypre should be on by default. It does seem like hypre is becoming the defacto default for nalu-wind...

Comment (psakievich) [reply configs/crusher/packages.yaml:102]: This is right. We want the default every where but AMD platforms to be `+trilinos-solvers+hypre` and on AMD platforms it should be `~trilinos-solvers+hypre`.

**Pull request merged**